src/core/server/plugins/plugins_system.test.ts

```diff
@@ -408,8 +408,8 @@ test('`uiPlugins` returns ordered Maps of all plugin manifests', async () => {
       "order-0",
       "order-1",
       "order-2",
-      "order-3",
       "order-4",
+      "order-3",
```

Comment:
Is the order change here expected?

Reply:
Yes, with this dependency graph, the order with sorted input
`[order0, order1, order2, order3, order4]`
and result will be 
`order0, order1, order2, order4, order3`


![image](https://github.com/user-attachments/assets/712f03f7-f780-44f6-bfee-d98aa1ba6317)

Reply:
As long as the order remains consistent, it should be fine. I'm just wondering why the order changed in this case, since localeCompare should place order3 before order4.

Reply:
If the two plugins are independent, the order of their setup will be the reverse of the order in the array. It's kind of similar with stack based on code here

https://github.com/opensearch-project/OpenSearch-Dashboards/blob/436d54d55e9ebfeaa0e2a6382ff8462bd167adf4/src/core/server/plugins/plugins_system.ts#L266-L267